dir/base.go

```diff
@@ -0,0 +1,94 @@
+package dir
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+const (
+	notation = "notation"
+)
+
+var (
+	// for mocking
+	goos          = runtime.GOOS
+	userConfigDir = os.UserConfigDir
+	userCacheDir  = os.UserCacheDir
+	getenv        = os.Getenv
+
+	// systemLibexec directory for binaries not meant to be executed directly by users' shell or scripts
```

Comment:
For the comment length, we should keep it under 80 characters per line except for URLs.
To set up the ruler check in VS Code: go to settings.json -> add `"editor.rulers": [ 80,],`
For detail, please refer to: [Readibility](https://teams.microsoft.com/l/message/19:142cc8ad20054170a602e7ba0e3d1a07@thread.skype/1656037597676?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=6cd5f724-f9f3-437c-ba5c-3316d1125c93&parentMessageId=1656037597676&teamName=acr-dev&channelName=ORAS%20Artifacts-OCI&createdTime=1656037597676)

Reply:
Thanks. Done.